src/lib/dfp/lazy-load.ts

```diff
@@ -35,37 +61,53 @@ const onIntersect = (
 	);
 };
 
-// Decide the rootMargin for the IntersectionObserver
-// This is based on the variant the user is in, control and
-// variant-20 are the same
-const decideLazyLoadMargin = (): string => {
-	const variant = getEagerPrebidVariant();
+const onIntersectPrebid = (
+	entries: IntersectionObserverEntry[],
+	observer: IntersectionObserver,
+) => {
+	const advertIds: string[] = [];
+	entries
+		.filter((entry) => !('isIntersecting' in entry) || entry.isIntersecting)
+		.forEach((entry) => {
+			log(
+				'commercial',
+				'prebid observer triggered for: ',
+				entry.target.id,
+			);
 
-	switch (variant) {
-		case 'variant-17':
-			return '17.5% 0px';
-		case 'variant-15':
-			return '15% 0px';
-		case 'variant-12':
-			return '12.5% 0px';
-		case 'variant-10':
-			return '10% 0px';
-		default:
-			return '20% 0px';
-	}
-};
+			observer.unobserve(entry.target);
+			requestBids(entry.target.id);
+			advertIds.push(entry.target.id);
+		});
 
-const getObserver = once(() => {
-	return Promise.resolve(
-		new window.IntersectionObserver(onIntersect, {
-			rootMargin: decideLazyLoadMargin(),
-		}),
+	dfpEnv.advertsToLoad = dfpEnv.advertsToLoad.filter(
+		(advert) => !advertIds.includes(advert.id),
 	);
```

Comment:
Just for my understanding, but here we're effectively marking these adverts as loaded at the point we run header bidding? Is it important we do this here, rather than doing it solely in the display ad observer?

Reply:
Oh interesting question, I didn't notice this was in both observers. I don't think will have any negative consequences as I don't think the `advertsToLoad` queue is ever used again after the page and ads have loaded.

I'll remove this duplicate though as it's just confusing!